### PR TITLE
Show what payments are accepted on user cards

### PR DESCRIPTION
**Pull request by mattbk**
Trying to implement #1262.

- [x] rough out logic for
    - [x] individuals
    - [x] teams
    - [x] organizations
- [x] add mouseover for icons (in particular, so people can see why no payment methods are available)
- [x] make it pretty
- [x] add to /donate pages
- [x] add to profile pages
- ~~write tests~~
- [ ] convince @Changaco

### CODE REVIEW

**Comment by mattbk** [comment]
Don't have a PayPal Glyphicon, so using PP for now. Need to adjust vertical alignment. 

screenshot from 2018-12-14 09-58-44

**Comment by Changaco** [comment]
Glyphicon doesn't have brand icons, we use Font Awesome for those (actually it's [Fork Awesome](https://forkawesome.github.io/Fork-Awesome/), but the in the code it's `fontawesome`). However your attempt to show payment methods as icons cannot succeed, because the SEPA Direct Debit payment method doesn't have a logo.

**Comment by mattbk** [comment]
Thanks @Changaco for the tip. 

If it's possible, one could submit a [SEPA logo](https://www.europeanpaymentscouncil.eu/document-library/other/sepa-logo-vector-format) to Fork Awesome, right?

**Comment by mattbk** [comment]
Another question. I copied the code from https://github.com/liberapay/liberapay.com/blob/7b9db6d877c4d6c7602580b3e05f67c946275ef8/www/%25username/donate.spt to show icons based on `payment_providers`, but how do I see if someone accepts SEPA?

**Comment by mattbk** [comment]
screenshot from 2018-12-14 12-17-30

**Comment by Changaco** [comment]
All users who have connected a valid Stripe account can receive SEPA direct debits, but only for donations in euro.

**Comment by mattbk** [comment]
> All users who have connected a valid Stripe account can receive SEPA direct debits, but only for donations in euro.

That seems like something that could be covered in a tooltip for all the "accepts credit card" accounts. Anything beyond that would increase profile box complexity.

**Comment by Changaco** [comment]
The SEPA logo doesn't fit well into Fork Awesome, it's too wide.

**Comment by mattbk** [comment]
I'd like to make it more immediately clear on /donate pages (e.g., https://liberapay.com/mattbk/donate) what payment options are available as well. Right now this is buries in the FAQ at the bottom of the page.

I would probably start with adding icons just above the "Next Step"/"Modify the donation" button, with the same `title` attributes.

**Comment by Changaco** [comment]
The addition to the profile boxes that you're proposing looks okay visually, but I'm not in favor of it.

1. It doesn't work for teams, because their boxes already have a third column showing the number of members (see <https://liberapay.com/explore/teams/>).
2. "Accepts" is imprecise and may be difficult to translate into other languages.
3. In general I don't think it's a good idea to show the available payment methods in profile boxes.

**Comment by Changaco** [comment]
> The SEPA logo doesn't fit well into Fork Awesome, it's too wide.

The special "€" sign in the SEPA logo would be used as a `sepa` icon in Fork Awesome, but it would be a copyright violation.

**Comment by mattbk** [comment]
> The special "€" sign in the SEPA logo could be used as a sepa icon in Fork Awesome, but it would be a copyright violation.

I think that would be very confusing, esp. for anyone who doesn't use EUR (like me).

The profile boxes expand for teams just like they do for individuals:

screenshot from 2018-12-17 09-17-11

**Comment by mattbk** [comment]
> "Accepts" is imprecise and may be difficult to translate into other languages.

I'm open to suggestions.

**Comment by mattbk** [comment]
Example on profile pages:

screenshot from 2018-12-19 12-52-18
screenshot from 2018-12-19 12-52-46

**Comment by mattbk** [comment]
I switched which side the logos are on for the previous buttons, but here is the /donate page as well.

screenshot from 2018-12-19 13-05-32

**Comment by mattbk** [comment]
Tests are next.

**Comment by tflex187** [comment]
Moneyyy https://www.paypal.me/TFlexon

**Comment by Changaco** [comment]
To unblock this PR I suggest using the `bank` icon to represent direct debits, and adding an appropriate tooltip covering all the icons (let's reuse the paragraph about payment methods displayed in the `/%username/donate` page).

I'm still not enthusiastic about displaying payment information in the Explore pages, I think it's trying to fit too much information in a small area and before the visitor has even shown any interest in actually donating. However it would be useful and appropriate to include the payment method icons in the `/%username/giving` page. Paradoxically that's the one page that this PR currently doesn't modify.

Don't worry about tests, this branch doesn't need any.

**Comment by mattbk** [comment]
Running into an issue where `title` tags are only showing the first word in a string.

File: templates/profile.html [hunk 1]
```diff
@@ -98,10 +99,28 @@ <h1>{{ participant.username }}</h1>
                         <a class="btn btn-donating btn-lg" href="{{ donate_path }}">{{
                             _("Donating ({amount})", amount=tip.amount)
                         }}</a>
-                    % else
-                        <a class="btn btn-donate btn-lg" href="{{ donate_path }}">
-                            {{ _("Donate") }}
-                        </a>
+                    % else                    
+                        <span title={{ accepts_summary }}><a class="btn btn-donate btn-lg" href="{{ donate_path }}">
```
Comment (Changaco) [inline templates/profile.html]: You forgot the quotes around `{{ accepts_summary }}`, it should be `title="{{ accepts_summary }}"`.

Comment (mattbk) [reply templates/profile.html]: Brilliant. I've made that mistake before!

**Comment by mattbk** [comment]
I got the logos to show up on /giving but not the tooltip yet.

image

**Comment by mattbk** [comment]
Titles work:

image

**Comment by Changaco** [comment]
Rebased on master and cleaned up.

I've moved the payment icons out of buttons, because it looks better to me that way and because it allows mobile users to tap on the icons to see the tooltip without risking to trigger the button.

**Comment by Changaco** [comment]
liberapay-2019-02-12-profile-with-payment-icons

**Comment by Changaco** [comment]
screenshot_20190212_150515

**Comment by Changaco** [comment]
screenshot_20190212_150148

**Comment by mattbk** [comment]
Nice. I have much to learn.

**Comment by mattbk** [comment]
I like the change to "Pledge" on people who don't have a payment method set up (see #1438), but do we need more explanation there of what that means?

image

Then on /donate *clicking through "Pledge"), is says "Donate" again with no explanation except for the buried "not ready to receive payments" paragraph below.

**Comment by Changaco** [comment]
screenshot_20190215_094846

**Comment by mattbk** [comment]
I like it. No additional changes on my end.

**Comment by trebmuh** [comment]
Can the "Wrong currency? Alice accepts 32 others:...". and the following be removed in case of a pledge?

The reasoning behind that is that it's probably confusing for a lambda user. This because, if Alice didn't configured any payment method, then how come can she accepts 32 other currencies?

**Comment by Changaco** [comment]
@trebmuh It might be a little confusing to some donors but I don't want to remove it. Most users only accept one currency, of course the switcher isn't displayed in that case.

@mattbk I've squashed your commits into a single one to keep the Git history "clean".

**Comment by trebmuh** [comment]
> @trebmuh It might be a little confusing to some donors but I don't want to remove it. Most users only accept one currency, of course the switcher isn't displayed in that case.

Ah ok, I thought that the currency switcher was still displayed in case of a user without any means of payment. All good then!

**Pull request merged**